src/niftyone/figures/generator.py

```diff
@@ -92,10 +107,27 @@ def __call__(
         overwrite: bool,
     ) -> None:
         # Filters by entity (via string query)
-        table_idxes = table.ent.query(self.query).index
-        for idx in table_idxes:
-            record = table.nested.loc[idx]
-            self.generate(record=record, out_dir=out_dir, overwrite=overwrite)
+        # First query is for main image, subsequent are for overlays
+        query_dfs = [table.ent.query(q) for q in self.queries]
+        indexed_dfs = [
+            (
+                df.assign(table_index=df.index)
```

Comment:
I think you can combine some of these steps with just `df.reset_index(names=f"index_{ii}")`?